### PR TITLE
 many: have install return encryption keys for data and save, improve tests

### PR DESCRIPTION
**Pull request by bboozzoo**
Refactor install.Run() to return state information about the installed system.
In particular, return the encryption/reovery keys for ubuntu-data and
ubuntu-save partitions. With this, the saving of keys has been moved to
devicestate, what allowed for further improvements in unit tests, to verify that
the keys are properly saved to disk.

Stacked on top of #9443. The last commit is relevant.

### CODE REVIEW

**Comment by bboozzoo** [comment]
This PR is no longer blocked. I've rebased it on top of master.

File: gadget/install/params.go [hunk 1]
```diff
@@ -31,9 +30,14 @@ type Options struct {
...
-	// at the end of the installation process.
-	ChosenEncryptionKey(secboot.EncryptionKey)
+// EncryptionKeySet is a set of encryption keys.
+type EncryptionKeySet struct {
+	Key         secboot.EncryptionKey
+	RecoveryKey secboot.RecoveryKey
+}
+
+// InstalledSystemState carries state information about an installed system.
+type InstalledSystemState struct {
```
Comment (pedronis) [inline gadget/install/params.go:40]: s/State/Info/ perhaps ?

File: overlord/devicestate/handlers_install.go [hunk 2]
```diff
@@ -197,13 +209,51 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		return fmt.Errorf("cannot make run system bootable: %v", err)
 	}
 
+	if useEncryption {
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:212]: we can/should do this earlier I think, inside the if trustedInstallObserver != nil

**Review by pedronis — COMMENTED** [review]
I did a pass, haven't looked closely at tests though

File: gadget/install/params.go [hunk 3]
```diff
@@ -31,9 +30,14 @@ type Options struct {
...
-	// ChosenEncryptionKey stores the encrypted data partition key to be sealed
-	// at the end of the installation process.
-	ChosenEncryptionKey(secboot.EncryptionKey)
+// EncryptionKeySet is a set of encryption keys.
+type EncryptionKeySet struct {
+	Key         secboot.EncryptionKey
+	RecoveryKey secboot.RecoveryKey
+}
+
+// InstalledSystemSideData carries the side data of an installed system.
```
Comment (pedronis) [inline gadget/install/params.go:39]: s/the side data/side data/

... of an installed system,  e.g.  secrets to access its partitions.

File: tests/lib/uc20-create-partitions/main.go [hunk 4]
```diff
@@ -91,13 +91,23 @@ func main() {
 		Mount:   args.Mount,
 		Encrypt: args.Encrypt,
 	}
-	err = installRun(args.Positional.GadgetRoot, args.Positional.Device, options, obs)
+	installState, err := installRun(args.Positional.GadgetRoot, args.Positional.Device, options, obs)
```
Comment (pedronis) [inline tests/lib/uc20-create-partitions/main.go:94]: s/installState/installSideData/ ?

**Review by pedronis — APPROVED** [review]
thanks, some small comments

**Review by cmatsuoka — APPROVED** [review]
Thanks Maciek, lgtm.

**Pull request merged**